joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/passes/ConfigFileCreationPass.scala

```diff
@@ -0,0 +1,95 @@
+package io.joern.javasrc2cpg.passes
+
+import better.files.File
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.NewConfigFile
+import io.shiftleft.passes.ConcurrentWriterCpgPass
+import io.shiftleft.utils.IOUtils
+import org.slf4j.LoggerFactory
+
+import java.nio.file.Paths
+import scala.util.{Failure, Success, Try}
+
+class ConfigFileCreationPass(projectDir: String, cpg: Cpg) extends ConcurrentWriterCpgPass[File](cpg) {
+
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  override def generateParts(): Array[File] = {
+    Try(File(projectDir)) match {
+      case Success(file) if file.isDirectory =>
+        file.listRecursively
+          .filter(isConfigFile)
+          .toArray
+
+      case Success(file) if isConfigFile(file) =>
+        Array(file)
+
+      case _ => Array.empty
+    }
```

Comment:
We have https://github.com/joernio/joern/blob/master/joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/SourceFiles.scala#L16 for this.

Reply:
That method would work for some of these, but `isConfigFile` is more complicated than a simple extension check. It might not be necessary to do the more complex checks, but I did it this way for the sake of consistency with java2cpg.